Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Definitions/FileTransporter.cs

```diff
@@ -5,461 +5,472 @@
-                                    {
-
-                                        userResultMessage = $"Can't trust SFTP server. The server fingerprint does not match. " +
-                                                               $"Expected fingerprint: '{_batchContext.Connection.ServerFingerPrint}', but was: '{BitConverter.ToString(e.FingerPrint).Replace("-", ":")}'";
-                                        // If previous failed try with MD5 typed fingerprint
-                                        var expectedFingerprint = Util.ConvertFingerprintToByteArray(_batchContext.Connection.ServerFingerPrint);
-                                        e.CanTrust = (e.FingerPrint.SequenceEqual(expectedFingerprint)) ? true : false;
-                                    }
-                                    
-                                };
-                            }
-                            catch (Exception e)
-                            {
-                                _logger.NotifyError(null, "Error when checking the server fingerprint: ", e);
-                                return FormFailedFileTransferResult(userResultMessage);
-                            }
-
-                        }
-                        client.ConnectionInfo.Timeout = TimeSpan.FromSeconds(_batchContext.Connection.ConnectionTimeout);
-
-                        client.BufferSize = _batchContext.Connection.BufferSize;
+                                    var sha256Fingerprint = Convert.ToBase64String(mySHA256.ComputeHash(e.HostKey));
+                                    // Set the userResultMessage in case checking server fingerprint failes.
+                                    userResultMessage = $"Can't trust SFTP server. The server fingerprint does not match. " +
+                                                            $"Expected fingerprint: '{_batchContext.Connection.ServerFingerPrint}', but was: '{sha256Fingerprint}'";
+                                    e.CanTrust = (sha256Fingerprint == _batchContext.Connection.ServerFingerPrint);
+                                }
 
-                        client.Connect();
+                                if (!e.CanTrust)
+                                {
 
-                        if (!client.IsConnected)
+                                    userResultMessage = $"Can't trust SFTP server. The server fingerprint does not match. " +
+                                                            $"Expected fingerprint: '{_batchContext.Connection.ServerFingerPrint}', but was: '{BitConverter.ToString(e.FingerPrint).Replace("-", ":")}'";
+                                    // If previous failed try with MD5 typed fingerprint
+                                    var expectedFingerprint = Util.ConvertFingerprintToByteArray(_batchContext.Connection.ServerFingerPrint);
+                                    e.CanTrust = e.FingerPrint.SequenceEqual(expectedFingerprint) ? true : false;
```

Comment:
`SequenceEqual` method return true or false, so you can just pass that result to `e.CanTrust` instead of the if-statements.

Reply:
Actually if I have understood correctly the e.CanTrust needs to be set in order to the fingerprint to work.

Reply:
Scratch that, I was on the wrong line. Fixed